src/main/java/com/couchbase/lite/store/ForestDBStore.java

```diff
@@ -1208,8 +1208,8 @@ private boolean saveForest(Document doc,
         if(!isWinner)
             // TODO
             ;
-        if (properties != null && properties.containsKey("type"))
-            doc.setType((String)properties.get("type"));
+        if (properties != null && properties.containsKey("type") && properties.get("type") instanceof String)
+            doc.setType((String) properties.get("type"));
```

Comment:
In cbforest, type is declared with alloc_slice which is extends from slice. slice's default value is NULL to buffer pointer and size is 0. I believe we don't need to set null here. 
https://github.com/couchbaselabs/cbforest/blob/87f93e8ab25fa653288fe019f13e8bbc1ad3bcf2/CBForest/slice.hh#L50